File: android/src/main/kotlin/com/mindcruiser/mc_flutter_recorder/RecorderException.kt
package com.mindcruiser.mc_flutter_recorder

class RecorderException(val err: ErrorType, message: String?) : Exception(message)

enum class ErrorType {
    ILLEGAL_ARGUMENT,
    ILLEGAL_STATE,
    NO_PERMISSION,
    NO_SPACE,
    INTERRUPTED,
    UNKNOWN,
    DURATION_EXCEEDED,
}

File: lib/recorder_exception.dart
class RecorderException implements Exception {
  final String message;
  final ErrorType errorType;

  RecorderException(this.errorType, [this.message = ""]);
}

enum ErrorType {
  illegalArgument,
  illegalState,
  noPermission,
  noSpace,
  interrupted,
  unknown,
  durationExceeded,
}
